modules/mocked_authentication/lib/mockdata/reader.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module MockedAuthentication
+  module Mockdata
+    class Reader
+      def self.find_credentials(credential_type:)
+        file_names = get_file_names(credential_type)
+        get_mocked_data(file_names)
+      end
+
+      class << self
+        private
+
+        def get_file_names(type)
+          file_directory = get_file_directory_for_type(type)
+          users = Dir.entries(file_directory).select { |file| validate_file_name(file) ? file : nil }
+          users.map { |file| "#{file_directory}/#{file}" }
+        end
+
+        def get_file_directory_for_type(type)
+          "#{Settings.sign_in.mock_credential_dir}/credentials/#{type}"
+        end
+
+        def validate_file_name(name)
```

Comment:
Oops, this was my bad, the mock data files should have a `.json`, do you mind just updating the `writer` in this PR to include `.json` when we write files?